src/celeritas/global/alongstep/detail/PropagationApplier.hh

```diff
@@ -127,17 +129,17 @@ PropagationApplierBaseImpl<MP>::operator()(CoreTrackView const& track)
             // machine epsilon compared to the actual position. This case seems
             // to happen mostly in vecgeom when "stuck" on a boundary, so it
             // may not lead to an infinite loop because the state is changing.
-            CELER_LOG(error)
+            CELER_LOG_LOCAL(error)
                 << "Propagation of step length " << repr(sim.step_length())
                 << " due to post-step action "
                 << sim.post_step_action().unchecked_get()
                 << " leading to distance " << repr(p.distance)
                 << (p.boundary  ? " (boundary hit)"
                     : p.looping ? " (**LOOPING**)"
                                 : "")
-                << " failed to change position at " << repr(orig_pos)
-                << " with ending direction "
-                << repr(track.make_geo_view().dir());
+                << " failed to change position";
+            sim.status(TrackStatus::errored);
+            sim.post_step_action(track.tracking_cut_action());
```

Comment:
Not yet I think: this is only in debug mode, it takes three extra full precision registers, and worst of all I know why it's usually happening: our field integration is taking place in global coordinates. (The integration sub-substep sizes are so small it's no wonder we step without changing position.) We should perhaps refactor field integration (from the substepper/integrator and lower, at least) so that we accumulate in a "from the start of the step" coordinate system, and then transform to the global system when calculating the local field.

Reply:
Ahh ok, good we understand what's causing it. If it's just in debug we don't really care about the extra registers, do we? Why not have the same behavior on host and device?

Reply:
I see what you mean. Sure, why not.